src/Dapr.PluggableComponents.Tests/Adaptors/PubSubAdaptorTests.cs

```diff
@@ -0,0 +1,379 @@
+            context);
+
+        Assert.NotNull(response);
+        Assert.Empty(response.Features);
+    }
+
+    [Fact]
+    public async Task FeaturesWithFeatures()
+    {
+        var mockStateStore = new Mock<IPubSub>();
+        var mockFeatures = mockStateStore.As<IPluggableComponentFeatures>();
+
+        mockFeatures
+            .Setup(component => component.GetFeaturesAsync(It.IsAny<CancellationToken>()))
+            .ReturnsAsync(new[] { "feature1", "feature2" });
+
+        var adaptor = AdaptorFixture.CreatePubSub(mockStateStore.Object);
+
+        using var context = new TestServerCallContext();
+
+        var response = await adaptor.Features(
+            new FeaturesRequest(),
+            context);
+
+        Assert.NotNull(response);
+        Assert.Equal(2, response.Features.Count);
+        Assert.Contains("feature1", response.Features);
+        Assert.Contains("feature2", response.Features);
+
+        mockFeatures
+            .Verify(
+                component => component.GetFeaturesAsync(
+                    It.Is<CancellationToken>(token => token == context.CancellationToken)),
+                Times.Once());
+    }
+
+    [Fact]
+    public async Task Publish()
+    {
+        var mockStateStore = new Mock<IPubSub>();
+
+        mockStateStore
+            .Setup(component => component.PublishAsync(It.IsAny<PubSubPublishRequest>(), It.IsAny<CancellationToken>()));
+
+        var adaptor = AdaptorFixture.CreatePubSub(mockStateStore.Object);
+
+        using var context = new TestServerCallContext();
+
+        string topic = "topic";
+
+        var response = await adaptor.Publish(
+            new PublishRequest
+            {
+                Topic = topic,
+            },
+            context);
+
+        mockStateStore
+            .Verify(
+                component => component.PublishAsync(
+                    It.Is<PubSubPublishRequest>(request => request.Topic == topic),
+                    It.Is<CancellationToken>(token => token == context.CancellationToken)),
+                Times.Once());
+    }
+
+    [Fact]
+    public async Task PullMessagesWithTopic()
+    {
+        var mockStateStore = new Mock<IPubSub>();
+
+        mockStateStore
+            .Setup(component => component.PullMessagesAsync(It.IsAny<PubSubPullMessagesTopic>(), It.IsAny<MessageDeliveryHandler<string?, PubSubPullMessagesResponse>>(), It.IsAny<CancellationToken>()));
+
+        var adaptor = AdaptorFixture.CreatePubSub(mockStateStore.Object);
+
+        using var context = new TestServerCallContext();
+
+        string topic = "topic";
+
+        var mockWriter = new Mock<IServerStreamWriter<PullMessagesResponse>>();
+
+        var reader = new PullMessagesRequestStreamReader();
+
+        var pullTask = adaptor.PullMessages(
+            reader,
+            mockWriter.Object,
+            context);
+
+        await reader.AddAsync(new PullMessagesRequest { Topic = new Topic { Name = topic } });
+
+        reader.Complete();
```

Comment:
Should this have a timeout?

Reply:
Makes sense; I've added one to the pull-related tests.